README.md

```diff
@@ -1,14 +1,102 @@
 
 ## Features
 
-This extension adds support for the following:
+The extension makes possible remote deployment of edited Qorus source files directly from the VSCode editor.
 
-- Remote development
-- Qorus objects templates
-- etc.
+## How to use
 
+### Configuration data and corresponding tree view
+
+In order to work with the extension a project must be open in the VSCode editor. By project we mean the root directory of a project (the root of a git repository, for a typical example).
+
+Next, in the root directory there needs to be a project configuration file named **qorusproject.json** defining development environments and Qorus instances used during the project development. If the configuration file **qorusproject.json** is not there it will be generated from a template so that the user can simply edit its contents seeing the required structure.
+
+The configuration file can be opened for editing either normally from the file explorer or using the context menu over each file and each folder within the project.
+
+Based on the configuration file contents a *Qorus instances tree view* is provided in the explorer pane. Switching to the *Qorus instances tree view* is done by clicking the hexagon [Q] icon in the activity bar.
+
+More projects (directories) can be open at the same time. When editing alternately files from different projects the *Qorus instances tree view* changes accordingly.
+
+The configuration file data structure is explained by the following example:
+
+    {
+        "qorus_instances": {
+            "local": [
+                {
+                    "url": "http://localhost:8001",
+                    "name": "main local Qorus instance",
+                    "custom_urls": []
+                },
+                {
+                    "url": "https://localhost:8011",
+                    "name": "local Qorus instance"
+                }
+            ],
+            "dev": [
+                {
+                    "url": "http://1.2.3.4:5678",
+                    "name": "dev 1",
+                    "custom_urls": [
+                        {
+                            "url": "sftp://abc@def/ghi",
+                            "name": "abc"
+                        },
+                        {
+                            "url": "sftp://xyz@uvw",
+                            "name": "xyz"
+                        }
+                    ]
+                },
+                {
+                    "url": "https://2.3.4.5:6789",
+                    "name": "dev 2"
+                }
+            ]
+        }
+    }
+
+The interpretation of the data is depicted by a tree with three levels:
+- development **environments** (here '*local*' and '*dev*')
+- - **Qorus instances** (here both environments have two Qorus instances)
+- - - **URLs**: the Qorus instance "main" URL at the first position and then the custom URLs, if any (here only the '*dev 1*' Qorus instance in the '*dev*' environment has any custom URLs)
```

Comment:
explanation of "custom urls" is missing.

The main goal of custom URLs is to provide an operating system integrated method how to provide a project related shortcuts for a developer.